app/services/pipeline/storage/assignment_summary_service.rb

```diff
@@ -3,27 +3,84 @@
+    #
+    # Expected context structure:
+    #   - context.assignment: The Assignment record
+    #   - context.student_feedbacks: Array of processed student work (optional)
+    #   - context.parsed_response: OpenStruct containing:
+    #     - qualitative_insights: String (class-wide analysis)
+    #     - feedback_items: Array of OpenStructs with:
+    #       - item_type: String ('strength' or 'opportunity')
+    #       - title: String
+    #       - description: String
+    #       - evidence: String
+    #
+    # Updates context with:
+    #   - context.saved_summary: The created AssignmentSummary with feedback items
+    #
+    # @example
+    #   context = Pipeline::Context::AssignmentSummary.new
+    #   context.assignment = assignment_record
+    #   context.student_feedbacks = processed_student_works
+    #   context.parsed_response = parsed_llm_response
+    #   result = Pipeline::Storage::AssignmentSummaryService.call(context: context)
+    #
     class AssignmentSummaryService
       def self.call(context:)
         Rails.logger.info("Storing summary for assignment: #{context.assignment.id}")
-        context.saved_summary = persist_to_database(context)
+
+        ActiveRecord::Base.transaction do
+          persist_summary(context)
+        end
+
         context
+      rescue => e
+        Rails.logger.error("Failed to store assignment summary: #{e.message}")
+        raise
       end
 
-      def self.persist_to_database(context)
-        # This would contain actual persistence logic in a real implementation
+      private
+
+      def self.persist_summary(context)
         assignment = context.assignment
+        parsed_response = context.parsed_response
 
-        if context.parsed_response.present?
-          summary = assignment.assignment_summary || AssignmentSummary.new(assignment: assignment)
-          summary.update!(
-            qualitative_insights: context.parsed_response[:qualitative_insights],
-            student_work_count: assignment.student_works.count
-          )
+        # Create assignment summary record
+        summary = AssignmentSummary.create!(
+          assignment: assignment,
+          qualitative_insights: parsed_response.qualitative_insights,
+          student_work_count: calculate_student_work_count(context)
+        )
+
+        # Create feedback items (class-wide strengths and opportunities)
+        create_feedback_items(summary, parsed_response.feedback_items)
 
-          summary
+        # Update context with saved summary
+        context.saved_summary = summary
+      end
+
+      def self.calculate_student_work_count(context)
+        # Use student_feedbacks from context if available, otherwise count from assignment
+        if context.student_feedbacks.present?
+          context.student_feedbacks.size
         else
-          assignment.assignment_summary
+          context.assignment.student_works.count
+        end
+      end
```

Comment:
I think it might be nice to ask the context how many student works it has. We should create a `student_work_count` method on the context, unit test it, and call it here instead of this implementation.